docs/spec/spec.md

```diff
@@ -0,0 +1,373 @@
+# Specification: Ballerina Redis Library
+
+_Authors_: @NipunaRanasinghe \
+_Reviewers_: @AyeshLK @AzeemMuzammil \
+_Created_: 2024/02/19 \
+_Updated_: 2024/02/28 \
+_Edition_: Swan Lake
+
+## Introduction
+
+This is the specification for the Redis connector library of the [Ballerina language](https://ballerina.io),
+which provides a comprehensive set of operations to interact with Redis servers using Ballerina.
+
+The Redis library specification undergoes continuous refinement, and any updates are documented and released under
+relevant GitHub tags.
+
+If you have any feedback or suggestions about the library, start a discussion via
+a [GitHub issue](https://github.com/ballerina-platform/ballerina-library/issues)
+or in the [Discord server](https://discord.gg/ballerinalang).
+Based on the outcome of the discussion, the specification and implementation can be updated. Community feedback is
+always welcome.
+
+Any accepted proposal, which affects the specification is stored under `/docs/proposals`. Proposals under discussion can
+be found with the label `type/proposal` in GitHub.
+
+The conforming implementation of the specification is released to Ballerina Central. Any deviation from the
+specification is considered a bug.
+
+## Contents
+
+1. [Overview](#1-overview)
+2. [Client](#2-client)  
+   2.1. [Client Initialization](#21-client-initialization)  
+   2.2. [Secure communication](#22-secure-communication)  
+   2.3. [Cluster connections](#23-cluster-connections)  
+   2.4. [Connection pooling](#24-connection-pooling)   
+   2.5. [Closing the client](#25-closing-the-client)  
```

Comment:
@NipunaRanasinghe shall we organize the content in following structure. This is the structure we followed in the other connectors.
```
2. Client
  2.1. Configurations
    - Secure socket
    - Connection URI and ConnectionParams
    - ConnectionConfig
  2.2 Initialization
  2.3 Secure communication
  2.4 Cluster connection
  2.5 Connection pooling
  2.6 Closing the client
  ```

Reply:
Addressed with https://github.com/ballerina-platform/module-ballerinax-redis/pull/219/commits/63431bc46cdcc29040c8ace4bd731891cc346189